### ISSUE TITLE
Add Semantic Version Policy from stylelint

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint-config-standard#86]
Should we add the _Semantic Version Policy_ from stylelint? (Adapted if needed) 

> https://github.com/stylelint/stylelint#semantic-versioning-policy?

**Comment by ntwb** [comment]
Draft text via https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress/issues/97

```md

## Semantic Versioning Policy

stylelint-config-standard follows [semantic versioning](http://semver.org). However, due to the nature of stylelint as a code quality tool, it's not always clear when a minor or major version bump occurs. To help clarify this for everyone, based on stylelint's [semantic versioning policy](https://github.com/stylelint/stylelint#semantic-versioning-policy) which is in turn based on ESLint's [semantic versioning policy](https://github.com/eslint/eslint#semantic-versioning-policy), we've defined the following subset of stylelint's semantic versioning policy for stylelint-config-standard:

-   Patch release (intended to not break your lint build)
    -   A bug fix in a rule that results in stylelint reporting fewer errors.
    -   Improvements to documentation.
    -   Non-user-facing changes such as refactoring code, adding, deleting, or modifying tests, and increasing test coverage.
    -   Re-releasing after a failed release (i.e., publishing a release that doesn't work for anyone).

-   Minor release (might break your lint build)
    -   A bug fix in a rule that results in stylelint reporting more errors.
    -   A new rule is added.
    -   A new option to an existing rule that does not result in stylelint reporting more errors by default.
    -   An existing rule is deprecated.

-   Major release (likely to break your lint build)
    -   A change in the documented behaviour of an existing rule results in stylelint reporting more errors by default.
    -   An existing rule is removed.

According to our policy, any minor update may report more errors than the previous release (ex: from a bug fix). As such, we recommend using the tilde (`~`) in `package.json` e.g. `"stylelint": "~7.2.0"` to guarantee the results of your builds.
```

The above is taken from the upcoming stylelint v8 semver policy and needs reviewing

**Comment by jeddy3** [comment]
> Should we add the Semantic Version Policy from stylelint? (Adapted if needed)

I don't think so. This package is independent of, and quite different to, the stylelint package. This package is a _config_, whereas stylelint is a _build tool_. The `~` approach, as [you said](https://github.com/stylelint/stylelint/issues/2756#issuecomment-315759730), is about a balance for build tools. For a config however, every addition is very likely to break someone's build and I suggest we stick to our current policy of a major release for any removal, change or addition.

**Comment by ntwb** [comment]
Thanks, makes sense 👍
